diff --git a/autogen.sh b/autogen.sh
index 9e068f3..16d781d 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -15,7 +15,7 @@
 #-------------------------------------------------------------------------
 
 
-MONGOC_VERSION=1.9.5
+MONGOC_VERSION=1.14.0
 JSONC_VERSION=0.13.1-20180305
 
 if [ "$#" -ne 1 ]; then
@@ -79,9 +79,12 @@ function install_json_lib
 function install_mongoc_driver
 {
 	cd mongo-c-driver
-	./configure --with-libbson=auto --enable-ssl
+	mkdir cmake-build
+	cd cmake-build
+	cmake -DENABLE_AUTOMATIC_INIT_AND_CLEANUP=OFF ..
+	make
 	make install
-	cd ..
+	cd ../..
 }
 
 ###
