diff --git a/src/multicorn.c b/src/multicorn.c
index a43d28e..2660572 100644
--- a/src/multicorn.c
+++ b/src/multicorn.c
@@ -119,7 +119,7 @@ _PG_init()
 	/* Make sure that the python interpreter is initialized by plpy
 	 * and not by Multicorn. */
 	SPI_connect();
-	SPI_execute("SELECT multicorn_check_plpython3u()", false, 0);
+	SPI_execute("SELECT public.multicorn_check_plpython3u()", false, 0);
 	SPI_finish();
 #endif
 	oldctx =  MemoryContextSwitchTo(CacheMemoryContext);
